pkg/sanity/node.go

```diff
@@ -451,6 +451,19 @@ var _ = DescribeSanity("Node Service", func(sc *TestContext) {
 		})
 
 		It("should remove target path", func() {
+			// This test may break for consumers that are using
+			// custom target path functions if they have not yet
+			// implemented similar functionality to check if the
+			// path exists. Skip this test if there is a custom
+			// command or function provided to create the path,
+			// but not yet provided to check the path.
+			if sc.Config.CreateTargetPathCmd != "" && sc.Config.CheckPathCmd == "" {
+				Skip("CreateTargetPathCmd was set, but CheckPathCmd was not. Please update your testing configuration to enable CheckPathCmd.")
+			}
+			if sc.Config.CreateTargetDir != nil && sc.Config.CheckPath == nil {
+				Skip("CreateTargetDir was set, but CheckPath was not. Please update your testing configuration to enable CheckPath.")
```

Comment:
Both should always be non-nil because they have defaults.

What you need to check for here is "sc.Config.CreateTargetDir not default and sc.Config.CheckPath not default".

Also, I think this only needs to be checked when `sc.Config.CreateTargetPathCmd` is unset. Otherwise we check and use `sc.Config.CheckPathCmd`.

Reply:
`createMountTargetLocation` has 2 separate paths, one for the custom function, but if that is nil then it calls mkdir. `sc.Config.CreateTargetDir` is nil by default.
https://github.com/dobsonj/csi-test/blob/a2091104af4d4e1df72bc6cfa8de4d955afa411a/pkg/sanity/sanity.go#L363-L378
Originally I did try to compare `sc.Config.CheckPath != defaultCheckPath` here, but got the following compile error:
```
pkg/sanity/node.go:463:63: invalid operation: sc.Config.CheckPath != defaultCheckPath (func can only be compared to nil)
```
Which is interesting... I think that should work fine in C, but apparently not in Go.
So I ended up changing `CheckPath` to follow the same model as `createMountTargetLocation` and leave `sc.Config.CheckPath` set to nil by default, specifically so the check on line 463 would work.
https://github.com/dobsonj/csi-test/blob/a2091104af4d4e1df72bc6cfa8de4d955afa411a/pkg/sanity/sanity.go#L539-L545

If `sc.Config.CreateTargetPathCmd` is set, we should hit the skip statement on line 461 and not make it down to line 464. Unless `CreateTargetPathCmd`, `CheckPathCmd`, and `CreateTargetDir` are all set at the same time, which seems strange. I can add a check for that though if that's what you're pointing out.

Reply:
> So I ended up changing CheckPath to follow the same model as createMountTargetLocation and leave sc.Config.CheckPath set to nil by default, specifically so the check on line 463 would work.

That's the part that I had missed. Then the current code is fine.